include/spi.h

```diff
@@ -386,6 +419,78 @@ static inline int _impl_spi_release(const struct spi_config *config)
 	return api->release(config);
 }
 
+#ifdef CONFIG_SPI_SLAVE
+/* SPI Slave part: */
+
+/**
+ * @brief Start a controller to act as an SPI slave, if supported
+ *
+ * Note: This function is synchronous.  If successful, given configuration
+ *       will be exclusive to the related controller until spi_slave_stop()
+ *       is called: controller cannot be switched from slave to master and
+ *       back to slave as long as a slave transaction is meant to happen.
+ *       That is valid only for controllers supporting both modes at runtime.
+ *
+ * @param config Pointer to a valid spi_config structure instance.
+ * @param tx_bufs Buffer array where data to be sent originates from,
+ *        or NULL if none.
+ * @param rx_bufs Buffer array where data to be read will be written to,
+ *        or NULL if none.
+ *
+ * @retval 0 If successful, negative errno code otherwise.
+ */
+static inline int spi_slave_transceive(const struct spi_config *config,
+				       const struct spi_buf_set *tx_bufs,
+				       const struct spi_buf_set *rx_bufs)
+{
+	const struct spi_driver_api *api = config->dev->driver_api;
+
+	return api->transceive(config, tx_bufs, rx_bufs);
+}
+
+/**
+ * @brief Update buffers on slave events.
+ *
+ * Note: use this function on SPI_SLAVE_EVENT_INPUT
+ *
+ * @param config Pointer to a valid spi_config structure instance, must be
+ *        the same that was used to read the master input.
+ * @param tx_bufs Buffer array where data to be sent originates from.
+ * @param rx_bufs Buffer array where data to be received from the master
+ *        should be stored in.
+ *
+ * @retval 0 If successful, negative errno code otherwise.
+ */
+static inline int spi_slave_update_buffers(const struct spi_config *config,
```

Comment:
I don't understand the point to have the same spi_slave_transceive & spi_slave_update_buffers which are the exact same as spi_transceive ? adding the events should be enough I think.

Reply:
It's just aliases, for clearer user code. But indeed, nothing prevents you to just use spi_transceive directly.

Reply:
In this case, why don't you use the spi_transceive() function to be able to drive it from an application ?

Reply:
I don't follow you. You can use spi_transceive as well on a spi slave application, I just made these aliases for clarity. I still have to finish testing slave support on dw, but on the sample side, with the slave cb it's obvious when you read those aliases what is being done.

Reply:
Inline aliases won't hurt anyway.

Reply:
Yes, actually these are like spi_write/spi_read. Better wording would be "helpers", instead of aliases. Maybe it's worth documenting it.